src/modules/files-storage/api/uc/wopi.uc.ts

```diff
@@ -0,0 +1,66 @@
+import {
+	AuthorizationClientAdapter,
+	AuthorizationContextBuilder,
+	AuthorizationContextParams,
+	AuthorizationContextParamsRequiredPermissions,
+} from '@infra/authorization-client';
+import { Logger } from '@infra/logger';
+import { FileRecordParentType, GetFileResponse } from '@modules/files-storage/domain';
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain/types';
+import { FileRecord } from '../../domain/file-record.do';
+import { FilesStorageService } from '../../domain/service/files-storage.service';
+import { SingleFileParams } from '../dto';
+import { WopiCheckFileInfoResponse } from '../dto/wopi.response';
+import { WopiResponseBuilder } from '../mapper';
+
+@Injectable()
+export class WopiUc {
+	constructor(
+		private readonly filesStorageService: FilesStorageService,
+		private readonly authorizationClientAdapter: AuthorizationClientAdapter,
+		private readonly logger: Logger
+	) {
+		this.logger.setContext(WopiUc.name);
+	}
+
+	public async checkFileInfo(params: SingleFileParams, userId: string): Promise<WopiCheckFileInfoResponse> {
+		const fileRecord: FileRecord = await this.filesStorageService.getFileRecord(params.fileRecordId);
+		const { parentId, parentType } = fileRecord.getProps();
+
+		const canWrite = await this.hasPermission(
+			parentType,
+			parentId,
+			AuthorizationContextBuilder.write([AuthorizationContextParamsRequiredPermissions.BOARD_EDIT]) // @TODO what permissions should be used here?
```

Comment:
We must find solutions to bypass parameters from vue frontend, over collabora to this service.
Based on this we can solve the userName, and figure out if we should request read, or write permissions. 
Based on the success request we return in wopi response read/write.